frontend/src/components/transactions/TransactionDetails.tsx

```diff
@@ -36,9 +39,30 @@ export default class extends React.Component<Props, State> {
         actionArgs = action[actionKind];
       }
       if (actionArgs.hasOwnProperty("deposit")) {
-        this.setState({ deposit: actionArgs.deposit });
+        return actionArgs.deposit;
+      } else {
+        return "0";
       }
     });
+    const IntLengths: Array<number> = deposits.map(dp => dp.length);
+    const max = Math.max(...IntLengths);
+    /* big.js for big int calculation is just adding number where the number is not "0", so like what two big int plus,
+      10001000 + 10004000, return [2,0,0,0,5] and the tail "0"s will gone. Remember the total length actual number is with
+      max
+    */
+    const bigIntDeposit = deposits
+      .map(dp => new Big(dp))
+      .reduce((current, dp) => dp.plus(current), 0)
+      .c.join("");
+    /* give back string "20005" */
+    if (bigIntDeposit.length < max) {
+      const tail = "0".repeat(max - bigIntDeposit.length);
+      const deposit = bigIntDeposit.concat(tail);
+      /* give back string "20005000" */
+      this.setState({ deposit });
+    } else {
+      this.setState({ deposit: bigIntDeposit });
+    }
```

Comment:
There is `.toString()` method for Big number. Why don't we use it?

Reply:
Actually .toString() will return "1.0000001e+25" instead of "10000000000000000000000000" which is not accepted by nearlib function for Balance component. We should use .toFixed(). I will change this.